presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -305,7 +306,166 @@ protected static ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaD
+                    log.info("Execute sql: %s", minMaxSql);
+
+                    statement.execute(minMaxSql);
+                    try (ResultSet rs = statement.getResultSet()) {
+                        rs.next();
+                        long minId = rs.getLong("MIN_VALUE");
+                        long maxId = rs.getLong("MAX_VALUE");
+                        int splitCount = deducePartitionCount(maxId - minId);
+
+                        if (splitCount > 1) {
+                            List<Pair<Long, Long>> partitions = partition(minId, maxId, splitCount);
+                            ranges = new ArrayList<>();
+                            for (Pair<Long, Long> x : partitions) {
+                                ranges.add(new JdbcSplit.RangeInfo(splitColumn.getName(), x.getKey(), x.getValue()));
+                            }
+
+                            // add a null range if the column is nullable
+                            if (splitColumn.isNullable()) {
+                                ranges.add(new JdbcSplit.RangeInfo(splitColumn.getName(), null, null));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        catch (SQLException e) {
+            log.error("getSplits failed!", e);
+        }
+
+        List<JdbcSplit> splits = new ArrayList<>();
+        if (ranges != null) {
+            for (JdbcSplit.RangeInfo range : ranges) {
+                splits.add(new JdbcSplit(Optional.empty(), Optional.of(range)));
+            }
+        }
+        else { // no split column
+            JdbcSplit jdbcSplit = new JdbcSplit(Optional.empty(), Optional.empty());
+            splits.add(jdbcSplit);
+        }
+
+        return new FixedSplitSource(splits);
+    }
+
+    /**
+     * Looking for the split column based on some predefined rules:
+     * 1. If there is a auto_increment column, pick it.
+     * 2. Otherwise if there is a bigint column, pick it.
+     * 3. Otherwise if there is an integer column, pick it.
+     * 4. Otherwise return null -- which means do not split
+     */
+    private ColumnInfo getSplitColumn(DatabaseMetaData metaData, String catalogName, String schemaName, String tableName)
+            throws SQLException
+    {
+        try (ResultSet resultSet = metaData.getColumns(catalogName, schemaName, tableName, null)) {
+            List<ColumnInfo> columns = new ArrayList<>();
+            while (resultSet.next()) {
+                String name = resultSet.getString("COLUMN_NAME");
+                String type = resultSet.getString("TYPE_NAME");
+                String isAutoIncrementStr = resultSet.getString("IS_AUTOINCREMENT");
+                boolean isAutoIncrement = "YES".equalsIgnoreCase(isAutoIncrementStr);
+
+                String isNullableStr = resultSet.getString("IS_NULLABLE");
+                boolean isNullable = "YES".equalsIgnoreCase(isNullableStr);
```

Comment:
`resultSet.getString("IS_NULLABLE")` is empty, the "nullability is unknown", so you should assume column to be nullable.

Generally, we already have this logic here:
https://github.com/prestosql/presto/blob/c6c1feb85f87701d07b04e334e8e99fb0306d41e/presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java#L256
(using `NULLABLE` instead of `IS_NULLABLE`)

So why don't you reuse the `getColumns` here?